src/ast/ast.c

```diff
@@ -25,11 +25,17 @@ static inline void _prepareIterateAll(rax *map, raxIterator *iter) {
 // referred_funcs will include: "abs", "max" and "min".
 static void _consume_function_call_expression(const cypher_astnode_t *expression,
 											  rax *referred_funcs) {
-	// Value is an apply operator
-	const cypher_astnode_t *func = cypher_ast_apply_operator_get_func_name(expression);
+	// Expression is an Apply or Apply All operator.
+	bool apply_all = (cypher_astnode_type(expression) == CYPHER_AST_APPLY_ALL_OPERATOR);
+
+	// Retrieve the function name and add to rax.
+	const cypher_astnode_t *func = (!apply_all) ? cypher_ast_apply_operator_get_func_name(expression) :
+								   cypher_ast_apply_all_operator_get_func_name(expression);
 	const char *func_name = cypher_ast_function_name_get_value(func);
 	raxInsert(referred_funcs, (unsigned char *)func_name, strlen(func_name), NULL, NULL);
 
+	if(apply_all) return;  // Apply All operators have no arguments.
+
```

Comment:
In which case `uint narguments = cypher_ast_apply_operator_narguments(expression);`
`narguments` should be 0 and we'll simply skip the following loop,

If that's indeed the case I prefer we remove this line and simply fall throw, one logic which works for all cases.

Reply:
Unfortunately, Apply All operators are parser objects of a different type and with a different interface than Apply operators (thus the conditional logic for name retrieval at https://github.com/RedisGraph/RedisGraph/pull/717/files#diff-51a289a0066db87d439f3b280ed18df4R32-R33).

This is to say that `cypher_ast_apply_operator_narguments` will cause a type failure when given an Apply All operator.